staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -226,23 +229,22 @@ func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 		return fmt.Errorf("--force-conflicts only works with --server-side")
 	}
 
-	if o.DryRun && o.ServerSideApply {
-		return fmt.Errorf("--dry-run doesn't work with --server-side (did you mean --server-dry-run instead?)")
+	if o.DryRunStrategy == cmdutil.DryRunClient && o.ServerSideApply {
+		return fmt.Errorf("--dry-run=client doesn't work with --server-side (did you mean --dry-run=server instead?)")
 	}
 
-	if o.DryRun && o.ServerDryRun {
-		return fmt.Errorf("--dry-run and --server-dry-run can't be used together")
+	if o.DryRunStrategy == cmdutil.DryRunClient && o.ServerDryRun {
+		return fmt.Errorf("--dry-run=client and --server-dry-run can't be used together")
```

Comment:
I think now you can make this error more useful, probably just append: `(did you mean --dry-run=server instead?)`

Reply:
Added